flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/streamrecord/LatencyMarker.java

```diff
@@ -67,31 +67,40 @@ public int getSubtaskIndex() {
 	// ------------------------------------------------------------------------
 
 	@Override
-	public boolean equals(Object o) {
-		if (this == o) {
+	public boolean equals(Object obj) {
```

Comment:
was the existing implementation wrong?

Reply:
Yes, the prior implementation did this:
`if (operatorId != that.operatorId) {`
instead of calling equals(), which resulted in a false if the object is equivalent, but not the exact same object (for example, when the bytestream is copied).

These changes are just from eclipse generating the equals and hashCode methods. I needed to do this so that the unit test would actually pass since it uses the equals method in it's assertion.